### DISCUSSION TITLE
Inner Cyclotron LED limit?

### DISCUSSION THREAD

**Discussion by CrisisFilms** [gpstar81/GPStar-proton-pack#260] [Q&A]
I have a 45 LED ring that will fit perfectly inside the 3d printed cyclotron I made, but I see multiple references to the maximum number being 35. It seems that there is the concern of having too many LEDs and causing delays, but considering there's an allowance for a 40 LED outer cyclotron but I only plan use the 20 LED Frutto mod, it seems like there should still be some room to work with. I attempted making some changes but being very new\unfamiliar with this I was unable to get it working for myself. Can anyone advise whether this can be changed, or if I should stick with the smaller 35 LED ring?

**Comment by gpstar81** [comment]
It should be ok, it would just require some slight tweaks to the code. 45 seems kinda big though?

**Comment by CrisisFilms** [comment]
Thanks, I bought a few different ones to see what would work best and found this one which has the LEDs more tightly packed than many others like the larger 40 LED ring:
https://www.amazon.com/gp/product/B08GPCB7K3/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
It's got a 120mm outer diameter, so using TacoBelli's cyclotron files, I modified things and made a recessed area in the base of the cake for it to sit. The circuit board is slightly overlapped by the inner wall, so the LEDs themselves are just inside the cake, sitting below the modified DragonWorkshop's interior I made with clear filament for the light to show through.

**Comment by nomakewan** [comment]
I know this is an old post but you should definitely stick with the smaller LED ring if you can.

If you want to use 40, you can compile your own version of the firmware that makes the following changes (we'll be replacing the 35 LED inner cyclotron LED ring code):

- In the Proton Pack's `Configuration.h` change `uint8_t i_inner_cyclotron_cake_num_leds` to `40`
- In the Proton Pack's `Serial.h` change the line `i_inner_cyclotron_cake_num_leds = 35;` to `i_inner_cyclotron_cake_num_leds = 40;`
- The end.

The voice prompt will say "35 Inner Cyclotron LEDs" still but it'll be 40.

Due to the sheer number of lights already on the pack, combined with all of the serial data it has to juggle, that's why we can't really support a whole bunch. It's already really pushing it as-is, and we can't really offer options that _require_ a user set a lower value on a different device in order to use a higher value on another.

Hope this helped!
